oap-ape/ape-native/src/reader.cc

```diff
@@ -318,24 +322,40 @@ int Reader::readBatch(int32_t batchSize, int64_t* buffersPtr_, int64_t* nullsPtr
     time += std::chrono::steady_clock::now() - start;
   }
 
-  if (aggExprs.size()) {
+  if (aggExprs.size() && rowsRet > 0) {  // if rows after filter is 0, no need to do agg.
     auto start = std::chrono::steady_clock::now();
+    int index = 0;
     for (int i = 0; i < aggExprs.size(); i++) {
       auto agg = aggExprs[i];
       if (typeid(*agg) == typeid(RootAggExpression)) {
         std::vector<int8_t> tmp(0);
-        rowsRet = agg->ExecuteWithParam(rowsToRead, buffersPtr, nullsPtr, tmp);
+        agg->ExecuteWithParam(rowsRet, buffersPtr, nullsPtr, tmp);
         auto result = std::dynamic_pointer_cast<RootAggExpression>(agg)->getResult();
         if (result.size() == 1) {
           aggResults[i].push_back(result[0]);
         } else {
           ARROW_LOG(DEBUG) << "Oops... why return " << result.size() << " results";
         }
+        // TODO: refactor. For 'avg' expression, it will return two elements, which are
+        // 'Sum' and 'Count' and 'Count' type should be int64. However, we didn't handle
+        // 'Count' expression here.
+        for (int j = 0; j < result.size(); j++) {
+          if (j == 1) {  // for `count` in `avg`
+            *((int64_t*)(buffersPtr_[index])) = result[j]->toInt64();
```

Comment:
Flink will not push down an `avg` expession.
It will push down `sum` and `count` for `avg`.
So for Flink, there is no such problem to parse mixed data in the same return column.

Reply:
Got it. I will try to parse `avg` to `sum` and `count` in JsonParser.